products.py

```diff
@@ -0,0 +1,54 @@
+from __future__ import print_function
+from ipywidgets import interact, interactive, fixed, interact_manual
+import ipywidgets as widgets
+import datetime
+
+
+class GenericProductsPrinter(widgets.Output):
+    def __init__(self):
+        super(GenericProductsPrinter, self).__init__()
+        self.slider_out = widgets.Output()
+        self.pretty_print_out = widgets.Output()
+        self.timestamp_out = widgets.Output()
+        self.slider = None
+        self.prods = []
+        self.index = 0
+        
+        with self:
+            display(self.pretty_print_out)
+            display(self.slider_out)
+            display(self.timestamp_out)
+
+    def append(self, prod):
+        self.prods.append((prod, str(datetime.datetime.now())))
```

Comment:
You should print the timestamp at a better format

Reply:
Like what for example?